src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -1154,10 +1196,88 @@ public async Task<string> GetMethodName(int methodId, CancellationToken token)
             commandParamsWriter.Write(methodId);
 
             using var retDebuggerCmdReader = await SendDebuggerAgentCommand(CmdMethod.GetNameFull, commandParamsWriter, token);
-            var methodName = retDebuggerCmdReader.ReadString();
-            return methodName.Substring(methodName.IndexOf(":")+1);
+            return retDebuggerCmdReader.ReadString();
         }
 
+        public async Task<string> GetPrettyMethodName(int methodId, bool isAnonymous, CancellationToken token)
+        {
+            using var commandParamsWriter = new MonoBinaryWriter();
+            commandParamsWriter.Write(methodId);
+
+            using var retDebuggerCmdReader = await SendDebuggerAgentCommand(CmdMethod.GetPrettyName, commandParamsWriter, token);
+            var type = (ElementType)retDebuggerCmdReader.ReadInt32();
+            switch (type)
+            {
+                case ElementType.GenericInst:
+                {
+                    var ret = retDebuggerCmdReader.ReadString();
+                    if (ret.IndexOf(':') is int index && index > 0)
+                        ret = ret.Substring(0, index);
+                    ret = regexForAsyncMethodName.Replace(ret, "$1");
+                    var numGenericTypeArgs = retDebuggerCmdReader.ReadInt32();
+                    var numGenericMethodArgs = retDebuggerCmdReader.ReadInt32();
+                    int numTotalGenericArgs = numGenericTypeArgs + numGenericMethodArgs;
+                    var genericArgs = new List<string>(capacity: numTotalGenericArgs);
+                    for (int i = 0; i < numTotalGenericArgs; i++)
+                    {
+                        var typeArgC = retDebuggerCmdReader.ReadString();
+                        typeArgC = regexForGenericArgs.Replace(typeArgC, "");
+                        genericArgs.Add(typeArgC);
+                    }
+                    var match = regexForGenericArgs.Match(ret);
+                    while (match.Success)
```

Comment:
You can create an `ArraySegment<int>` from `genericArgs`. then instead of changing the list, and using Linq's Take, you can use `arr_seg.Slice(.., countArgs)`. It will avoid the extra allocations.